cartoframes/dataset.py

```diff
@@ -33,59 +33,80 @@ class Dataset(object):
     PUBLIC = 'public'
     LINK = 'link'
 
+    TABLE_TYPE = 'Table'
+    QUERY_TYPE = 'Query'
+    DATAFRAME_TYPE = 'DataFrame'
+    GEODATAFRAME_TYPE = 'GeoDataFrame'
+
     DEFAULT_RETRY_TIMES = 3
 
-    def __init__(self, query, data_type, context=None, bounds=None, schema='public', df=None):
+    def __init__(self, context=None, table_name=None, schema='public', query=None, df=None, geodf=None):
```

Comment:
What should be the expected behavior if someone inputs `table_name` and/or `query` and/or `df`, etc.? Maybe an error that only one should be specified at a time?

Reply:
^^ @oleurud 

I proposed to use generic parameters like `data` and `type`. I think this could be a solution to avoid multi-input types.

Reply:
We can not use this way. For example:
- With a dataset created by a query, the user wants to play with the data. We need at least `query` and `df`
- The user wants to upload a dataset created by a local DataFrame: we need at least `df` and `table name`
    ...

It is also a problem to extend functionality in the future.

I understand the problem, but I think it is not a solution. My point here is that we would only document the class methods and allowing an advanced user to add several parameters knowing what is doing (for example adding a local DataFrame and an also a table name, which will we used to upload the data in CARTO)